substrate/frame/people/src/lib.rs

```diff
@@ -0,0 +1,2059 @@
+		///
+		/// Parameters:
+		/// - `account`: The account to set the alias for.
+		/// - `call_valid_at`: The block number when the call becomes valid.
+		#[pallet::call_index(4)]
+		pub fn set_alias_account(
+			origin: OriginFor<T>,
+			account: T::AccountId,
+			call_valid_at: BlockNumberFor<T>,
+		) -> DispatchResultWithPostInfo {
+			let rev_ca = Self::ensure_revised_personal_alias(origin)?;
+			let now = frame_system::Pallet::<T>::block_number();
+			let time_tolerance = Self::account_setup_time_tolerance();
+			ensure!(
+				call_valid_at <= now && now <= call_valid_at.saturating_add(time_tolerance),
+				Error::<T>::TimeOutOfRange
+			);
+			ensure!(T::AccountContexts::contains(&rev_ca.ca.context), Error::<T>::InvalidContext);
+			ensure!(!AccountToPersonalId::<T>::contains_key(&account), Error::<T>::AccountInUse);
+
+			let old_account = AliasToAccount::<T>::get(&rev_ca.ca);
+			let old_rev_ca = old_account.as_ref().and_then(AccountToAlias::<T>::get);
+
+			let needs_revision = old_rev_ca.is_some_and(|old_rev_ca| {
+				old_rev_ca.revision != rev_ca.revision || old_rev_ca.ring != rev_ca.ring
+			});
+
+			// Ensure it changes the account associated, or it needs revision.
+			ensure!(
+				old_account.as_ref() != Some(&account) || needs_revision,
+				Error::<T>::AliasAccountAlreadySet
+			);
+
+			// If the old account is different from the new one:
+			// * decrease the sufficients of the old account
+			// * increase the sufficients of the new account
+			// * check new account is not already in use
+			if old_account.as_ref() != Some(&account) {
+				ensure!(!AccountToAlias::<T>::contains_key(&account), Error::<T>::AccountInUse);
+				if let Some(old_account) = &old_account {
+					frame_system::Pallet::<T>::dec_sufficients(old_account);
+					AccountToAlias::<T>::remove(old_account);
+				}
+				frame_system::Pallet::<T>::inc_sufficients(&account);
+			}
+
+			AccountToAlias::<T>::insert(&account, &rev_ca);
+			AliasToAccount::<T>::insert(&rev_ca.ca, &account);
+
+			if old_account.is_none() || needs_revision {
+				Ok(Pays::No.into())
+			} else {
+				Ok(Pays::Yes.into())
+			}
+		}
+
+		#[pallet::call_index(5)]
+		pub fn unset_alias_account(origin: OriginFor<T>) -> DispatchResult {
+			let alias = Self::ensure_personal_alias(origin)?;
+			let account = AliasToAccount::<T>::take(&alias).ok_or(Error::<T>::InvalidAccount)?;
+			AccountToAlias::<T>::remove(&account);
+			frame_system::Pallet::<T>::dec_sufficients(&account);
+
+			Ok(())
+		}
+
+		#[pallet::call_index(6)]
+		pub fn reset_root(origin: OriginFor<T>) -> DispatchResultWithPostInfo {
+			ensure_root(origin)?;
+			let _: Vec<_> = Root::<T>::drain().collect();
+			let _: Vec<_> = Keys::<T>::drain().collect();
+			let _: Vec<_> = People::<T>::drain().collect();
+			let _: Vec<_> = AliasToAccount::<T>::drain().collect();
+			let _: Vec<_> = AccountToAlias::<T>::drain().collect();
+
+			Ok(Pays::No.into())
+		}
+
```

Comment:
This was only for testing no, otherwise we should limit the number of removal and do accurate benchmark.
```suggestion
```

Reply:
Yes, I thought we could keep it until we release but no need actually.